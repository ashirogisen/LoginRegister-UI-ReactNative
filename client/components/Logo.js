import React from 'react'
import { StyleSheet, View, Image } from 'react-native'

function Logo() {
  const styles = StyleSheet.create({
    image: {
      width: 330,
      height: 100
    }
  })

  return (
    <View testID="test-logoContainer">
      <Image testID="test-logoImage" style={styles.image} source={require('../assets/logo.png')}></Image>
    </View>
  )
}

export default Logo