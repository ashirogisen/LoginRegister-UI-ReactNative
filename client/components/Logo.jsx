import React from 'react'
import { StyleSheet, View, Image } from 'react-native'

function Logo() {
  const styles = StyleSheet.create({
    image: {
      width: 330,
      height: 100
    }
  })

  return (
    <View>
      <Image style={styles.image} source={require('../assets/logo.png')}></Image>
    </View>
  )
}

export default Logo