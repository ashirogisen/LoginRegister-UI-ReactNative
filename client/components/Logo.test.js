import React from 'react'
import Logo from './Logo'
import renderer from 'react-test-renderer';
import { render } from 'react-native-testing-library';

describe('Render Logo', () => {
  it('Renders without crashing', () => {
    const tree = renderer.create(
      <Logo />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render a container', () => {
    const { getByTestId } = render (
      <Logo />
    )
    expect(getByTestId('test-logoContainer')).toBeTruthy()
  })

  it('Should render a logo image', () => {
    const { debug, getByTestId } = render (
      <Logo />
    )
    // debug()
    expect(getByTestId('test-logoImage')).toBeTruthy()
  })
})
