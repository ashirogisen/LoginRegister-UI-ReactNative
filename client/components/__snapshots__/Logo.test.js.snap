// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Logo Renders without crashing 1`] = `
<View
  testID="test-logoContainer"
>
  <Image
    source={1}
    style={
      Object {
        "height": 100,
        "width": 330,
      }
    }
    testID="test-logoImage"
  />
</View>
`;
